stl/inc/xmemory

```diff
@@ -776,6 +776,30 @@ _NODISCARD constexpr allocation_result<typename allocator_traits<_Alloc>::pointe
 // The number of user bytes a single byte of ASAN shadow memory can track.
 _INLINE_VAR constexpr size_t _Asan_granularity = 8;
 
+_NODISCARD inline const void* _Get_asan_aligned_first(const void* const _First, const void* const _Last) noexcept {
+    const auto _First_address = reinterpret_cast<uintptr_t>(_First);
+    const auto _Last_address  = reinterpret_cast<uintptr_t>(_Last);
+
+    const auto _Aligned_address = (_First_address + _Asan_granularity - 1) & ~(_Asan_granularity - 1);
+
+    if (_Aligned_address > _Last_address) {
+        return nullptr;
+    } else {
+        return reinterpret_cast<const char*>(_Aligned_address);

```

Comment:
Any reason for `const char*` here when the function returns `const void*`?

Reply:
missed edit.